### PR TITLE
Add action to build and push image to dockerhub&github packages

### PR DESCRIPTION
**Pull request by yuyulei**

### CODE REVIEW

**Comment by yuyulei** [comment]
TODO: use https://github.com/marketplace/actions/github-tag to get tag

File: .github/workflows/build-and-push-image.yml [hunk 1]
```diff
@@ -0,0 +1,252 @@
+name: Publish Docker image to Dockerhub Registry and GitHub Package Registry
+
+on:
+  push:
```
Comment (fatedier) [inline .github/workflows/build-and-push-image.yml:4]: only run when we publish a release?

File: .github/workflows/build-and-push-image.yml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+jobs:
+  binary:
+    name: Build Golang project
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.14
```
Comment (fatedier) [inline .github/workflows/build-and-push-image.yml:18]: 1.15?

File: .github/workflows/build-and-push-image.yml [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        id: go
+      -
+        name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      -
+        name: Get dependencies
```
Comment (fatedier) [inline .github/workflows/build-and-push-image.yml:24]: We don't need this step.

**Review by fatedier — APPROVED** [review]

**Pull request merged**